plugins/callback/terraform_stream.py

```diff
@@ -0,0 +1,68 @@
+# (C) 2020, Turntabl, <info@turtabl.io>
+# (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    callback: callback plugin to receive log streams
+    type: locking
+    short_description: creates a daemonized UDP socket server
+    description:
+      - Creates a socket server that is daemonized and  receives logs from terraform module.
+        The IP to stream the logs over should be set as an environment variable with name `ANSIBLE_TERRAFORM_STREAM_PORT`.
+        It should be the IP of the control node.
+        The port to stream over as well should be provided as environment variable with name `ANSIBLE_TERRAFORM_STREAM_HOST`.
+    requirements:
+     - Whitelist in configuration
+     - A writeable /var/log/ansible/hosts directory by the user executing Ansible on the controller
+'''
+import time
+import sys
+import os
+import re
+import socket
+import multiprocessing
+import random
+from ansible import constants as C
+from ansible import context
+from ansible.module_utils.common._collections_compat import MutableMapping
+from ansible.parsing.ajson import AnsibleJSONEncoder
+from ansible.plugins.callback import CallbackBase
+
+
+def sample_server(display_self):
+    server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

```

Comment:
We might be able to do this without explicit configuration.  If you bind to any available port on the local host here, you can then pass the port you get, to the code that writes.  I believe that binding to port 0 will give you an 'available' port, which you can find with a getsockname() call.  I think you can use INADDR_ANY for the hostname in the socket bind, to bind to all local addresses.  The writer then just has to send to the IP of the controller which you should be able to find in code.